src/components/lab/Lattice.js

```diff
@@ -27,22 +30,50 @@ class Lattice extends Component {
   };
 
   componentDidMount() {
-    // set up general animation for positive ions
-    setInterval(() => {
-      const {
-        temperature,
-        scales: { from },
-      } = this.props;
-      const { angle } = this.state;
-      const oscillation = IONS_OSCILLATION_RADIUS * Math.sin(angle);
+    const { isPaused } = this.props;
+    if (!isPaused) {
+      this.beginInterval();
+    }
+  }
+
+  componentDidUpdate({ isPaused: prevIsPaused }) {
+    const { isPaused } = this.props;
+    if (isPaused !== prevIsPaused && isPaused) {
+      clearInterval(this.interval);
+      this.interval = null;
+    } else if (isPaused !== prevIsPaused && !isPaused) {
+      this.beginInterval();
+    }
+  }
 
-      this.setState({
-        oscillation,
-        angle: angle + (temperature - from) * IONS_OSCILLATION_SPEED_FACTOR,
-      });
-    }, SET_INTERVAL_TIME);
+  componentWillUnmount() {
+    clearInterval(this.interval);
   }
 
+  beginInterval = () => {
+    if (!this.interval) {
+      // set up general animation for positive ions
+      this.interval = setInterval(() => {
+        const {
+          temperature,
+          scales: { from, to },
+        } = this.props;
+        const { angle } = this.state;
+        const oscillation = IONS_OSCILLATION_RADIUS * Math.sin(angle);
+
+        this.setState({
+          oscillation,
+          angle:
+            // transform from scale to float
+            // add an offset to allow oscillation even at minumum temperature
+            angle +
+            (((temperature + IONS_OSCILLATION_OFFSET - from) * 1.0) / to) *
+              IONS_OSCILLATION_SPEED_FACTOR,
```

Comment:
I believe the `/ to` here has caused the problem that the charges are no longer oscillating (the resulting angle is too small?). Not sure why it was changed from previous commits, but this needs revision.

Also, why the `* 1.0`?

Reply:
`*1.0` was to transform the initial temperature to a float. Which will result in a float after the division of the integer `to`. But I changed it to include the float part in the constant.

I revise a bit the computation, but the idea was to use `/to` to normalize the result, then have a more "neutral" speed factor.